File: 新东方CCCJunior/lecture5/FractionAction.java
import java.util.Scanner;

public class FractionAction {

	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);
		
		System.out.println("Numerator");
		int numerator = sc.nextInt();
		
		System.out.println("Denominator");
		int denominator = sc.nextInt();
		
		int quotient = numerator / denominator;
		System.out.print(quotient);

		numerator -= quotient * denominator;
		//55 - 5 * 10 == 0 
		//5
		int a = denominator;
		int b = numerator;
		if(numerator != 0) {
			//euclidean algorithm 辗转相除法
			//GCD(numerator denominator)
			int remainder;
			do {
				remainder = a % b;
				a = b;
				b = remainder;
			}while(remainder != 0);

			int gcd = a;
			numerator /= gcd;
			denominator /= gcd;
			
			System.out.print(" " + numerator + "/" + denominator);
		}
	}
}

File: prev_exams_and_ans/2002ccc/q4_fraction_action/fraction_action.py
numerator = int(input())
denominator = int(input())

quotient = int(numerator / denominator)
numerator -= quotient * denominator

if quotient != 0:
    print(quotient, " ", end='')

a = denominator
b = numerator

if (numerator != 0):
    while True:
        remainder = a % b
        a = b
        b = remainder

        if (remainder == 0):
            break
    gcd = a
    numerator /= gcd
    denominator /= gcd
        
    print(str(int(numerator)) + "/" + str(int(denominator)))

